flang/lib/Common/Fortran.cpp

```diff
@@ -134,8 +135,12 @@ bool AreCompatibleCUDADataAttrs(std::optional<CUDADataAttr> x,
     } else {
       if (*x == CUDADataAttr::Device) {
         if ((y &&
-                (*y == CUDADataAttr::Managed || *y == CUDADataAttr::Unified)) ||
+                (*y == CUDADataAttr::Managed || *y == CUDADataAttr::Unified ||
+                    *y == CUDADataAttr::Shared)) ||
             (!y && (isCudaUnified || isCudaManaged))) {
+          if (y && *y == CUDADataAttr::Shared) {
+            *warning = "SHARED attribute ignored"s;
```

Comment:
are you sure that `warning` is not a null pointer at this point in the code?  if not, please make this assignment conditional.  if it can't ever be a null pointer, then please transform it into a reference.

Reply:
Done.